twitter/twitter_client.bal

```diff
@@ -0,0 +1,1057 @@
+//
+// Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+import ballerina/http;
+import ballerina/io;
+
+string clientIdStr = "";
+string clientSecretStr = "";
+string accessTokenStr = "";
+string accessTokenSecretStr = "";
+
+# Twitter Client object.
+#
+# + httpClient - HTTP Client.
+# + userClient - Twitter user client.
+# + trendsClient - Twitter trends client.
+public type TwitterClient client object {
+
+    http:Client httpClient;
+    TwitterUserClient|error userClient;
+    TwitterTrendsClient|error trendsClient;
+
+    public function __init(TwitterConfiguration twitterConfig) returns error? {
+        self.httpClient = new(TWITTER_API_URL, config = twitterConfig.clientConfig);
+        var isValid = verifyCredentials(self.httpClient, twitterConfig.clientId, twitterConfig.clientSecret,
+            twitterConfig.accessToken, twitterConfig.accessTokenSecret);
+        if (isValid is error) {
+            return isValid;
+        }
+        self.userClient = new(self);
+        self.trendsClient = new(self);
```

Comment:
Shouldn't errors returned when new-ing up user or trends clients also be returned on new-ing up `TwitterClient`? 
Here the user might get misled since we are just setting the result even if it is an error by making userClient and trendsClient to be of union types that include error.

I think we should change L35 and L36 as follows and return the errors here if an error is returned on calling `new`:
```ballerina
TwitterUserClient userClient;
TwitterTrendsClient trendsClient;
```